src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Concat.txt

```diff
@@ -12,6 +15,10 @@
 >> Concat(["A", "B", "C"], ThisRecord.Value)
 "ABC"
 
+// Explicit 'ThisRecord', with separator
```

Comment:
Some more tests to consider:

As with separator?
Interpolated string as separator?
Nested concats with separator?
Lambda reference is invalid in separator arg

Reply:
- As with separator: added
- Interpolated as separator: added
- Nested concat with separator: already there (Concat(ForAll(["A","B"] As X,Concat(ForAll(["C","D"] As Y, X.Value & Y.Value),Value, "-")),Value, ", "))
- Lambda reference invalid in separator: added